src/boot_O2/printutils.c

```diff
@@ -1,5 +1,12 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/printutils/PrintUtils_VPrintf.s")
+void PrintUtils_VPrintf(PrintCallback* pfn, const char* fmt, va_list args) {
+    _Printf(*pfn, pfn, fmt, args);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/printutils/PrintUtils_Printf.s")
+void PrintUtils_Printf(PrintCallback* pfn, const char* fmt, ...) {
+    va_list args;
+    va_start(args, fmt);
+
+    PrintUtils_VPrintf(pfn, fmt, args);
+}
```

Comment:
Shouldn't this kind of variadic functions have a `va_end` after it processed the arguments?
```suggestion
void PrintUtils_Printf(PrintCallback* pfn, const char* fmt, ...) {
    va_list args;
    va_start(args, fmt);

    PrintUtils_VPrintf(pfn, fmt, args);
    
    va_end(args);
}
```